### ISSUE TITLE
coc.nvim conflict with vim-go

### ISSUE THREAD

**Issue by fannheyward** [neoclide/coc.nvim#53]
**Describe the bug**
coc conflicts with vim-go completion.

**To Reproduce**
1. minimal init.vim

```
call plug#begin()
Plug 'neoclide/coc.nvim', {'do': 'yarn install'}
Plug 'fatih/vim-go', { 'for': 'go' }
call plug#end()
```

2. reproduce, `nvim main.go`:

```
package main

import "log"

func main() {

}
```

* A: move L6, input `log.` will trigger completion popup, and completion items is correct. It's OK.
* B: move L6, **save action first**, then input `log.` also trigger popup, but with wrong items.
* C: `CocRestart` will fix this.

**Expected behavior**
A

**Screenshots**
no

**Desktop (please complete the following information):**
 - Terminal: iTerm2 3.2
 - Neovim 0.3.1
 - Coc: 0.0.11

**Additional context**

**Comment by chemzqm** [comment]
`save action first` what is it means?

**Comment by fannheyward** [comment]
Move L6, `:w`, then start input.

**Comment by chemzqm** [comment]
I can reproduce it now, the syncnize not working well with gocode.

**Comment by chemzqm** [comment]
There's some bug with vim-go and neovim, the using of vim-go makes buffer update events not fired at all.

**Comment by chemzqm** [comment]
```
  let g:go_fmt_autosave = 0
```
could solve this problem.

**Comment by chemzqm** [comment]
Figured out it's caused by the `rename` function, I have to reattach the buffer if it's abandoned by rename.

**Comment by fannheyward** [comment]
Tested with dev branch, it has fixed. Thanks for your work.

**Comment by venjiang** [comment]
NVIM v0.6.1 and latest vim-go, coc.nvim 
The same problem occurred。

**Comment by chemzqm** [comment]
Should be fixed by 3dc25e03
